### PR TITLE
Add instructions for running integration tests

### PR DESCRIPTION
**Pull request by sigalsax**
### What does this PR do?
This PR adds step by step instructions for how to run integration tests

### What ticket does this PR close?
Resolves -

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: docs/instruction_for_testing.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+It is **important** to note that this guide is not strictly for running tests while prepping for a release. This guide should be reviewed and the testing process defined here should be run a few weeks before a release, as you incrementally add logic. That way, if bugs are uncovered, you have enough time to address them!
+
+## Deployment and Setup
+
+Before we can run our integration tests, we need to deployment the required environments and create the necessary artifacts.
+
+At a high level, you will need to:
+
+1. Deploy and configure machines
+1. Clone the CLI/SDK repository
```
Comment (sigalsax) [inline docs/instruction_for_testing.md:14]: TODO - clone the repo in the new machines

File: docs/instruction_for_testing.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+Congrats! You ran project's integration tests.
+
+## Running tests with configurable environment
+
+As part of our test suite, we also want to test the CLI against different types of Conjur server environments. We want to validate the flow when the CLI makes a request to a server and the certificate configured is either 1. a known, verifiable CA-signed certificate or 2. a non-verifiable CA-signed certificate.
+
+Python comes with a list of trusted CAs. Every certificate signed by those CAs will be considered trusted certificate. Those CAs are located in a file called `cacerts.pem`. To get the path to this file, run in `python3 -c "import certifi; print(certifi.where())"`
```
Comment (sigalsax) [inline docs/instruction_for_testing.md:62]: TODO - will be considered a trusted and verifiable

File: docs/instruction_for_testing.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+At a high level, you will need to:
+
+1. Pick a platform to run the additional test flow (Hint: macOS should be the easiest)
+1. Complete the walk-through doc  (TODO ADD LINK) for configuring a certificate starting at the *Build CLI with your CA included* section.
+1. Once you have built the `integrations_tests_runner` with a verifiable CA-signed certificate, run the `integrations_tests_runner` executable one last time. The two tests mentioned in the previous section should fail.
+
```
Comment (sigalsax) [inline docs/instruction_for_testing.md:69]: TODO - Remove the certificate from the cacert.pem file and rerun again

File: docs/instruction_for_testing.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+## Miscellaneous tests
+
+There are a couple of manual checks that need to be done in additional to the regular integration tests.
+
+You should compress the *signed* Windows CLI executables (not the `integrations_test_runner`!) and run it on a Windows machine. 
+
+This will verify two things:
+
+1. The executable doesn't get corrupted during compression/decompression
```
Comment (sigalsax) [inline docs/instruction_for_testing.md:80]: TODO - The final deliverable is a `zip` so this ensures that the executable doesn't get corrupted during compression/decompression

File: docs/instruction_for_testing.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+## Miscellaneous tests
+
+There are a couple of manual checks that need to be done in additional to the regular integration tests.
+
+You should compress the *signed* Windows CLI executables (not the `integrations_test_runner`!) and run it on a Windows machine. 
+
+This will verify two things:
+
+1. The executable doesn't get corrupted during compression/decompression 
+1. The executable is not flagged as a virus by popular Anti-Virus softwares. For Windows machines, you can build the CLI (not the `integrations_test_runner`!) and upload the file to [Virustotal](https://www.virustotal.com/gui/) to verify. In addition to this inital check, copy over the CLI to a Windows machine and run it. Running `conjur init` should suffice.
```
Comment (sigalsax) [inline docs/instruction_for_testing.md:81]: TODO - should suffice. If the CLI does not trigger an Anti-Virus, you should be good.

File: docs/instruction_for_testing.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
+# Running tests before release
```
Comment (sigalsax) [inline docs/instruction_for_testing.md:1]: TODO - change filename to instructions_for_running_integration_tests
Title How to run full integration test suite before release

File: docs/instruction_for_testing.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+At a high level, you will need to:
+
+1. Deploy and configure machines
+1. Clone the CLI/SDK repository
+1. Install dependencies on those machines
+1. Build the `integrations_tests_runner`
+
+#### Deploy and configure machines
+
+You will need a machine to host your Conjur server (should be Appliance) and 4 different EC2 machines (Windows, RHEL 7, RHEL 8, macOS - if available) for packing the executable. If macOS is not available, your local macOS machine should suffice.
```
Comment (sigalsax) [inline docs/instruction_for_testing.md:20]: TODO - add clone the repository on those machines. You will need the project to build the artifact used for testing

File: docs/instruction_for_testing.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+#### Install dependencies
+
+On those 4 machines, clone the project's repostiory and install dependencies. To install dependencies,  follow the instructions [here](https://github.com/cyberark/conjur-api-python3/blob/master/CONTRIBUTING.md#development).
+
+#### Build the `integrations_tests_runner`
+
+The test runner executable wraps the CLI and the integration tests into a single executable, allowing us to run our integration tests without installing additional dependencies except for what the CLI requires. For more information see [here](https://github.com/cyberark/conjur-api-python3/blob/master/CONTRIBUTING.md#running-tests-outside-of-a-containerized).
```
Comment (mbenita-Cyberark) [inline docs/instruction_for_testing.md:27]:
```suggestion
The test runner executable wraps the CLI and the integration tests into a single executable, allowing us to run our integration tests without installing additional dependencies (**including python**) except for what the CLI requires. For more information see [here](https://github.com/cyberark/conjur-api-python3/blob/master/CONTRIBUTING.md#running-tests-outside-of-a-containerized). 
```

File: docs/instruction_for_testing.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
+# How to run full integration test suite
+
+The purpose of this document is to detail the steps to run the project's integration tests on each supported platform. Not all test flows have been integrated in our automated pipeline so it is important to follow this guide and understand how to run tests in a semi-automatically fashion.
+
+It is **important** to note that this guide is not strictly for running tests while prepping for a release. This guide should be reviewed and the testing process defined here should be run before a release, as you incrementally add logic. That way, if bugs are uncovered, you have enough time to address them!
```
Comment (eladkug) [inline docs/instruction_for_testing.md:5]: Not only before a release

File: docs/instruction_for_testing.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+## Deployment and Setup
+
+Before we can run our integration tests, we need to deploy the required environments and create the test artifact.
+
+At a high level, you will need to:
+
+1. Deploy and configure machines
+1. Install dependencies
```
Comment (eladkug) [inline docs/instruction_for_testing.md:14]: Create development environment and install requirements

File: docs/instruction_for_testing.md (hunk 8 above)
Comment (eladkug) [reply docs/instruction_for_testing.md:27]: Remove the link since it exist in this doc as well.

File: docs/instruction_for_testing.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+For the non-verifiable CA-signed certificate flow, you will need to run through the flow described in this section again. All you need to do is:
+
+1. Remove the cached directories.
+1. Remove the CA pem from the `cacerts.pem`.
+1. Rebuild the `integrations_tests_runner`. 
+
+Note that for this flow, two tests mentioned in the previous section *should pass*.
+
+## Miscellaneous tests
```
Comment (eladkug) [inline docs/instruction_for_testing.md:112]: The following tests should be run before release only (enough time to fix new issues)

File: docs/instruction_for_testing.md (hunk 9 above)
Comment (sigalsax) [reply docs/instruction_for_testing.md:5]: Done

File: docs/instruction_for_testing.md (hunk 10 above)
Comment (sigalsax) [reply docs/instruction_for_testing.md:14]: Done

File: docs/instruction_for_testing.md (hunk 7 above)
Comment (sigalsax) [reply docs/instruction_for_testing.md:19]: Done

File: docs/instruction_for_testing.md (hunk 8 above)
Comment (sigalsax) [reply docs/instruction_for_testing.md:27]: Done

File: docs/instruction_for_testing.md (hunk 11 above)
Comment (sigalsax) [reply docs/instruction_for_testing.md:112]: Done

**Review by mbenita-Cyberark — APPROVED** [review]

**Pull request merged**